File: April/mumbling.py
# Mumbling

# CodeWars Link - https://www.codewars.com/kata/5667e8f4e3f572a8f2000039

# Problem Statement
# You are given a string, return the a string with each index of the original string
# captailized, followed by lowercase of that index, with the amount depending on the index

# Examples
# accum("abcd") -> "A-Bb-Ccc-Dddd"
# accum("RqaEzty") -> "R-Qq-Aaa-Eeee-Zzzzz-Tttttt-Yyyyyyy"
# accum("cwAt") -> "C-Ww-Aaa-Tttt"


# My Solution
# learned strings can be filled with the same value by using *int
def accum(s):

    myList = []
    counter = 0

    for item in s:
        myList.append(item.upper() + item.lower()*counter)
        counter += 1

    return "-".join(myList)


# Better Solution
# learned about this easy syantax for a single line of a for loop
# if the for loop repeats only one line, consider using this sytanx
def accum(s):
    return '-'.join(c.upper() + c.lower() * i for i, c in enumerate(s))

File: July/Mumbling.js
// Mumbling

// CodeWars Link - https://www.codewars.com/kata/5667e8f4e3f572a8f2000039/javascript

// Problem Statement
// Create a function that creates a index repetition patterned string given a input string

// Example
// accum("abcd") -> "A-Bb-Ccc-Dddd"
// accum("RqaEzty") -> "R-Qq-Aaa-Eeee-Zzzzz-Tttttt-Yyyyyyy"
// accum("cwAt") -> "C-Ww-Aaa-Tttt"

// My Solution
function accum(s) {
    let myArr = s.split('');

    for (let i = 0; i < myArr.length; i++) {
        myArr[i] = myArr[i].toUpperCase() + myArr[i].toLowerCase().repeat(i);
    }

    return myArr.join('-');
}